src/NodaTime.Test/DateIntervalTest.cs

```diff
@@ -208,5 +208,274 @@ public void Deconstruction()
                 Assert.AreEqual(end, actualEnd);
             });
         }
+
+        [Test]
+        public void Contains_NullInterval_Throws()
+        {
+            var start = new LocalDate(2017, 11, 6);
+            var end = new LocalDate(2017, 11, 10);
+            var value = new DateInterval(start, end);
+
+            Assert.That(() => value.Contains(null), Throws.TypeOf<ArgumentNullException>());
```

Comment:
Any reason not to use `Assert.Throws<ArgumentException>(() => value.Contains(null))`? That's what we typically do... I only use `Assert.That` where it adds significant value, otherwise it feels like fluff to me.

Reply:
No reason in particular, it's just the way I'm more used to, so I kind of do it this way automatically. I'm going to change it :)